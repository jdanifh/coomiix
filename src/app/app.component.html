<cmx-layout [title]="title">
  <cmx-side-menu menu></cmx-side-menu>
  <router-outlet content></router-outlet>
</cmx-layout>
