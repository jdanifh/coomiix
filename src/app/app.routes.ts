import { Routes } from '@angular/router';

export const routes: Routes = [
  {
    path: '**',
    title: 'Home',
    loadComponent: () => import('./views/home/home.component').then(mod => mod.HomeComponent)
  },
];
