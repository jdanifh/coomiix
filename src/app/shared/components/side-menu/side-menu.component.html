<a routerLink="/" routerLinkActive="active" matRipple>
  <mat-icon aria-hidden="true">home</mat-icon>
  <span>Home</span>
</a>
<a routerLink="/dashboard" routerLinkActive="active" matRipple>
  <mat-icon aria-hidden="true">dashboard</mat-icon>
  <span>Dashboard</span>
</a>
