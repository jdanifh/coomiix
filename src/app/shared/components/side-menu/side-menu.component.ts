import { ChangeDetectionStrategy, Component } from '@angular/core';
import { RouterModule } from '@angular/router';
import { MatButtonModule } from '@angular/material/button';
import { MatIconModule } from '@angular/material/icon';
import { MatRippleModule } from '@angular/material/core';

@Component({
  selector: 'cmx-side-menu',
  imports: [
    RouterModule,
    MatIconModule,
    MatButtonModule,
    MatRippleModule,
  ],
  templateUrl: './side-menu.component.html',
  styleUrl: './side-menu.component.scss',
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class SideMenuComponent {

}
