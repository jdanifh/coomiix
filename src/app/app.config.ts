import { ApplicationConfig, provideExperimentalZonelessChangeDetection, provideZoneChangeDetection } from '@angular/core';
import { provideAnimationsAsync } from '@angular/platform-browser/animations/async';
import { provideRouter, withDebugTracing, withInMemoryScrolling, withViewTransitions } from '@angular/router';
import { provideHttpClient, withInterceptors } from '@angular/common/http';
import { provideLocales } from './core/providers/locale/locale.provider';

import { routes } from './app.routes';

// Zone.js is disabled in favor of zoneless change detection
export const appConfig: ApplicationConfig = {
  providers: [
    provideExperimentalZonelessChangeDetection(),
    provideAnimationsAsync(),
    provideRouter(routes, withDebugTracing(), withViewTransitions(), withInMemoryScrolling()),
    provideHttpClient(withInterceptors([])),
    provideLocales(),
  ]
};
