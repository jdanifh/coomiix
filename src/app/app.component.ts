import { Component, ChangeDetectionStrategy } from '@angular/core';
import { RouterOutlet } from '@angular/router';
import { LayoutComponent } from "./shared/components/layout/layout.component";
import { SideMenuComponent } from "./shared/components/side-menu/side-menu.component";

@Component({
  selector: 'app-root',
  standalone: true,
  changeDetection: ChangeDetectionStrategy.OnPush,
  imports: [
    RouterOutlet,
    LayoutComponent,
    SideMenuComponent
  ],
  templateUrl: './app.component.html',
  styleUrl: './app.component.scss'
})
export class AppComponent {
  public title = 'Coomiix';
}
