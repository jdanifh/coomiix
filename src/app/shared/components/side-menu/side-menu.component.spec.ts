import { ComponentFixture, TestBed } from '@angular/core/testing';

import { SideMenuComponent } from './side-menu.component';
import { provideRouter } from '@angular/router';
import { routes } from '../../../app.routes';
import { By } from '@angular/platform-browser';

describe('SideMenuComponent', () => {
  let component: SideMenuComponent;
  let fixture: ComponentFixture<SideMenuComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [SideMenuComponent],
      providers: [provideRouter(routes)]
    })
    .compileComponents();

    fixture = TestBed.createComponent(SideMenuComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should render Home button', () => {
    const home = fixture.debugElement.children[0].query(By.css('span'));
    expect(home.nativeElement.textContent).toBe('Home');
  });

  it(`should match snapshot`, () => {
    expect(fixture.nativeElement.innerHTML).toMatchSnapshot();
  })

});
