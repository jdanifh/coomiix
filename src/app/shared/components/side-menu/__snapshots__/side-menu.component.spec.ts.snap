// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideMenuComponent should match snapshot 1`] = `"<a routerlink="/" routerlinkactive="active" matripple="" class="mat-ripple" ng-reflect-router-link="/" ng-reflect-router-link-active="active" href="/"><mat-icon role="img" aria-hidden="true" class="mat-icon notranslate material-icons mat-ligature-font mat-icon-no-color" data-mat-icon-type="font">home</mat-icon><span>Home</span></a><a routerlink="/dashboard" routerlinkactive="active" matripple="" class="mat-ripple" ng-reflect-router-link="/dashboard" ng-reflect-router-link-active="active" href="/dashboard"><mat-icon role="img" aria-hidden="true" class="mat-icon notranslate material-icons mat-ligature-font mat-icon-no-color" data-mat-icon-type="font">dashboard</mat-icon><span>Dashboard</span></a>"`;
